include/seastar/net/tls.hh

```diff
@@ -198,9 +198,39 @@ namespace tls {
          * https://gnutls.org/manual/html_node/Priority-Strings.html
          *
          * Allows specifying order and allowance for handshake alg.
+         *
+         * @note: This function is a no-op when using OpenSSL.  Use
+         * set_cipher_string and set_ciphersuites
          */
         void set_priority_string(const sstring&);
 
+        /**
+         * Used to set the cipher string for TLS versions 1.2 and below
+         *
+         * @note: This function is a no-op when using GnuTLS.  Use
+         * set_priority_string
```

Comment:
Is it worth conditionally enabling these functions based on `SEASTAR_WITH_TLS_OSSL`, resulting in a compile-time failure when switching to/from OpenSSL, rather than using an empty set of ciphers that silently "breaks" TLS?

Reply:
that's probably fair.  I was trying to avoid doing that (I think we can agree that `#ifdef` is pretty ugly).